app/models/supported_ssg.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require 'yaml'
+
+# rubocop:disable Metrics/BlockLength
+SupportedSsg = Struct.new(:id, :package, :version, :profiles,
+                          :os_major_version, :os_minor_version,
+                          keyword_init: true) do
+  self::SUPPORTED_FILE = Rails.root.join('config/supported_ssg.yaml')
+
+  class << self
+    def all
+      raw_supported['supported'].map do |rhel, values|
+        major, minor = os_version(rhel)
+
+        new(
+          id: rhel,
+          os_major_version: major,
+          os_minor_version: minor,
+          **map_attributes(values)
+        )
+      end
+    end
+
+    def revision
+      raw_supported['revision']
+    end
+
+    private
+
+    def map_attributes(values)
+      attrs = values.slice(*members.map(&:to_s))
```

Comment:
Slightly easier to use (but optional):

```suggestion
      attrs = values.slice(*members.map(&:to_sym))
```

Reply:
`members` are symbols already. 
```
> SupportedSsg.members.map(&:to_sym)
=> [:id, :package, :version, :profiles, :os_major_version, :os_minor_version]
```

The keys in `values` are strings, therefore the conversion. Symbolizing is done after sanitation.